src/ansys/turbogrid/core/multi_blade_row/multi_blade_row.py

```diff
@@ -0,0 +1,1138 @@
+        progress_updates_queue,
+        report_stats_angle_unit,
+        report_mesh_quality_measures,
+        report_stats_decimal_places,
+    )
+    pyturbogrid_instance.save_state(filename=blade + ".tst")
+    pyturbogrid_instance.save_mesh(filename=blade + ".def")
+    pyturbogrid_instance.quit()
+    stop_dt = dt.now()
+    delta_dt = stop_dt - start_dt
+    delta_dt = td(seconds=int(delta_dt / td(seconds=1)))
+    deldt_parts = str(delta_dt).split(":")
+    progress_updates_queue.put([bladerow + "/" + blade, f"Stop time: {stop_dt}"])
+    progress_updates_queue.put(
+        [
+            bladerow + "/" + blade,
+            f"Duration: {deldt_parts[0]} hours {deldt_parts[1]} minutes {deldt_parts[2]} seconds",
+        ]
+    )
+    progress_updates_queue.put([bladerow + "/" + blade, "Done"])
+
+
+def execute_tginit_bladerow(
+    tginit_file,
+    blade_row,
+    blade,
+    settings,
+    progress_updates_queue,
+    report_stats_angle_unit,
+    report_stats_decimal_places,
+    report_mesh_quality_measures,
+):
+    start_dt = dt.now()
+    progress_updates_queue.put([blade_row + "/" + blade, f"Starting {blade_row} producer"])
+    progress_updates_queue.put([blade_row + "/" + blade, f"Start time: {start_dt}"])
+
+    pyturbogrid_instance = launch_turbogrid(
+        product_version="24.1.0",
+        log_level=pyturbogrid_core.PyTurboGrid.TurboGridLogLevel.CRITICAL,
+        log_filename_suffix="_" + blade,
+    )
+    progress_updates_queue.put([blade_row + "/" + blade, f"pyturbogrid instance created"])
+
+    progress_updates_queue.put([blade_row + "/" + blade, f"read_tginit:: {tginit_file}"])
+    pyturbogrid_instance.read_tginit(path=tginit_file, bladerow=blade_row)
+
+    progress_updates_queue.put([blade_row + "/" + blade, f"unsuspend"])
+    pyturbogrid_instance.unsuspend(object="/TOPOLOGY SET")
+    progress_updates_queue.put([blade_row + "/" + blade, f"Applying meshing settings"])
+    for setting in settings:
+        pyturbogrid_instance.set_obj_param(setting[0], setting[1])
+    write_mesh_report(
+        blade_row,
+        blade,
+        pyturbogrid_instance,
+        progress_updates_queue,
+        report_stats_angle_unit,
+        report_mesh_quality_measures,
+        report_stats_decimal_places,
+    )
+    pyturbogrid_instance.save_state(filename=blade + ".tst")
+    pyturbogrid_instance.save_mesh(filename=blade + ".def")
+    pyturbogrid_instance.quit()
+    stop_dt = dt.now()
+    delta_dt = stop_dt - start_dt
+    delta_dt = td(seconds=int(delta_dt / td(seconds=1)))
+    deldt_parts = str(delta_dt).split(":")
+    progress_updates_queue.put([blade_row + "/" + blade, f"Stop time: {stop_dt}"])
+    progress_updates_queue.put(
+        [
+            blade_row + "/" + blade,
+            f"Duration: {deldt_parts[0]} hours {deldt_parts[1]} minutes {deldt_parts[2]} seconds",
+        ]
+    )
+    progress_updates_queue.put([blade_row + "/" + blade, "Done"])
+
+
+def execute_ndf_blade_row_ansys_labs(
```

Comment:
Do we want to combine the functions together? It would reduce duplicated code, and also allow to include a simple 'containerized' mode, so that we could test in github as well.

Reply:
There are four alternate methods here: execute_ndf_bladerow, execute_tginit_bladerow,  execute_tginit_blade_row_ansys_labs and execute_ndf_blade_row_ansys_labs. The portion mostly common to all are are already in a reused method write_mesh_report below. The rest of the portions in these methods are mainly about launching PyTurboGrid and connecting to it is in the Ansys Labs methods. Other portions of the four methods are somewhat duplicate but are just the book-keeping lines for adding messages to progress updates queue etc. I think merging these methods will add many if else blocks and make the single method very long.